src/wazuh_db/helpers/wdb_global_helpers.c

```diff
@@ -1181,3 +1182,67 @@ int* wdb_get_agents_ids_of_current_node(const char* connection_status, int *sock
 
     return array;
 }
+
+wdbc_result wdb_parse_chunk_to_json(char* input, cJSON** output_json, char** last_group_hash) {
+    char* payload = NULL;
+
+    wdbc_result status = wdbc_parse_result(input, &payload);
+    if (status == WDBC_OK || status == WDBC_DUE) {
+        cJSON* response = cJSON_Parse(payload);
+        int array_size = cJSON_GetArraySize(response);
+        if (response && array_size > 0) {
+            if (output_json) {
+                cJSON_AddItemToArray(*output_json, response);
```

Comment:
Does this work with arrays of multiple items?

Reply:
Yes, because` 'response'` is a cJSON array ( `[{"group":"group1,group2","group_hash":"ef48b4cd"}....]` ) so here I am adding the entire array.
The final format of 'output_json' is` [ [ {"group":"group1,group2","group_hash":"ef48b4cd"}, {"group":"group3,group4","group_hash":"abcdef"} ... ],[ {"group":"group3,group1","group_hash":"eeeeefffd"}, {"group":"group4,group5","group_hash":"3432cdef"}... ], ... ]`